src/librustc/middle/stability.rs

```diff
@@ -577,7 +579,18 @@ impl<'a, 'gcx, 'tcx> TyCtxt<'a, 'gcx, 'tcx> {
                 format!("{}", message)
             };
 
-            self.lint_node(lint, id, span, &msg);
+            let mut diag = self.struct_span_lint_node(lint, id, span, &msg);
```

Comment:
if only we had a `HirId` here, we could use [`struct_span_lint_hir`](https://github.com/rust-lang/rust/blob/d30b99f9c23f8e1d6ef993cc94da96510ad709b3/src/librustc/ty/context.rs#L2881) (#50928)